### DISCUSSION TITLE
NuxtJs is very slow running on windows environments

### DISCUSSION THREAD

**Discussion by guilherme-codes** [nuxt/nuxt#29181] [Sharing]
Specifications: Windows 11, Node 20, 16GB, Intel i7.
It occurs for both WSL Ubuntu and Pure Windows system.

I'm using a nuxt 3 module to load my design system library to my main application. And I'm working with a pnpm monorepo. It takes more than 10 seconds to load my components and render it.

When I try it on Macbook Air M1 works very fast.

![image](https://github.com/user-attachments/assets/3d288671-0db5-442a-b786-cec9bcaf0602)

**Comment by geminigeek** [comment]
try this, do not use localhost:3000 on windows its slow , use 127.0.0.1:3000

**Comment by guilherme-codes** [comment]
### Sharing the solution here.

To keep running it in localhost without performance problems just change the dev script in the package.json file to 
**nuxt dev --no-fork --host 127.0.0.1**

-------------

**Other solution can be:**

Go into your hosts file "c:\windows\system32\drivers\etc\hosts" (open it as an administrator in notepad) and add an entry for

127.0.0.1 localhost

Make sure it's not commented out and make sure that the ipv6 one "IS COMMENTED OUT"

This slowness happenes on Windows 11 because by default the localhost loopback resolution defaults to the ipv6 loopback and it has to timeout before it resolves to 127.0.0.1, it tries ipv6 first...

So comment out the ipv6 localhost in the hosts file and make sure the 127.0.0.6 isn't. It should look something like this

```
# localhost name resolution is handled within DNS itself.
127.0.0.1       localhost
# ::1             localhost
# Added by Docker Desktop
192.168.0.185 host.docker.internal
192.168.0.185 gateway.docker.internal
# To allow the same kube context to work on the host and the container:
127.0.0.1 kubernetes.docker.internal
# End of section
```

Additionall you should disable the loopback protection on your system, you can do that with this powershell

```
New-ItemProperty -Path HKLM:\System\CurrentControlSet\Control\Lsa\ -Name DisableLoopbackCheck -PropertyType DWORD -Value 1 -Force | Out-Null
```

Fixing your hosts file for localhost and disabling the loopback check should resovle the issue and allow localhost to be fast again.
